lib/galaxy/tools/wrappers.py

```diff
@@ -475,34 +474,14 @@ def __getattr__(self, key: Any) -> Any:
         if self.false_path is not None and key == "file_name":
             # Path to dataset was rewritten for this job.
             return self.false_path
-        elif key == "extra_files_path":
+        elif key in ("extra_files_path", "files_path"):
+            if not self.compute_environment:
+                # Only happens in WrappedParameters context, refactor!
+                return self.unsanitized.extra_files_path
             if self.__io_type == "input":
-                path_rewrite = self.compute_environment and self.compute_environment.input_extra_files_rewrite(
-                    self.unsanitized
-                )
-            else:
-                path_rewrite = self.compute_environment and self.compute_environment.output_extra_files_rewrite(
-                    self.unsanitized
-                )
-            if path_rewrite:
-                return path_rewrite
+                return self.compute_environment.input_extra_files_rewrite(self.unsanitized)
```

Comment:
This looks nice - I spent weeks years ago trying to work on files_path vs extra_files_path, inside and outside Pulsar- it was a mess and I probably made it a lot messier - thanks for taking this on.

Reply:
This iteration also took me about 2 weeks ... I hope we got it right this time, I'm cautiously optimistic.